extensions/amp-analytics/0.1/events.js

```diff
@@ -447,21 +477,144 @@ export class TimerEventTracker extends EventTracker {
         'Bad maxTimerLength specification');
     const callImmediate = 'immediate' in timerSpec ?
         Boolean(timerSpec['immediate']) : true;
+    const timerStart = 'startSpec' in timerSpec ? timerSpec['startSpec'] : null;
+    user().assert(!timerStart || typeof timerStart == 'object',
+        'Bad timer start specification');
+    const timerStop = 'stopSpec' in timerSpec ? timerSpec['stopSpec'] : null;
+    user().assert((!timerStart && !timerStop) || typeof timerStop == 'object',
+        'Bad timer stop specification');
+
+    const timerId = this.generateTimerId_();
+    const timerParams = {};
+    timerParams[this.TIMER_PARAMS_.INTERVAL_LENGTH] = interval;
+    timerParams[this.TIMER_PARAMS_.MAX_TIMER_LENGTH] = maxTimerLength;
+    timerParams[this.TIMER_PARAMS_.CALL_IMMEDIATE] = callImmediate;
+    this.trackers_[timerId] = timerParams;
+    if (!timerStart) {
+      this.startTimer_(timerId, eventType, listener);
+      this.trackers_[timerId][this.TIMER_PARAMS_.CAN_RESTART] = false;
+    } else {
+      this.trackers_[timerId][this.TIMER_PARAMS_.CAN_RESTART] = true;
+      const startTracker = trackerProvider(timerStart);
+      user().assert(startTracker, 'Cannot track timer start');
+      const startTrackerBuilder = startTracker.add.bind(startTracker, context,
+          timerStart['on'], timerStart,
+          this.handleTimerToggle_.bind(this, timerId, eventType, listener),
+          trackerProvider);
+      this.trackers_[timerId][this.TIMER_PARAMS_.START_BUILDER] =
+          startTrackerBuilder;
+      const unlistenStart = startTrackerBuilder();
+      this.trackers_[timerId][this.TIMER_PARAMS_.UNLISTEN_START] =
+          unlistenStart;
+      const stopTracker = timerStop ? trackerProvider(timerStop) : null;
+      if (!!stopTracker) {
+        const stopTrackerBuilder = stopTracker.add.bind(stopTracker, context,
```

Comment:
same here, are we going to deal with user error from `#stopTracker.add()`?

Reply:
Same as above.